src/form_routes.py

```diff
@@ -0,0 +1,161 @@
+'''
+Copyright (C) 2020 Link Shortener Authors (see AUTHORS in Documentation).
+Licensed under the MIT (Expat) License (see LICENSE in Documentation).
+'''
+import uuid
+
+from sanic import Blueprint
+from sanic.response import redirect, html, json
+
+from sanic_oauth.blueprint import login_required
+
+from sanic_wtf import SanicForm
+
+from wtforms import StringField, SubmitField
+from wtforms.validators import DataRequired
+
+from models import actives, inactives
+
+
+form_blueprint = Blueprint('forms')
+
+
+class CreateForm(SanicForm):
+    endpoint = StringField('Endpoint', validators=[DataRequired()])
+    url = StringField('URL', validators=[DataRequired()])
+    submit = SubmitField('Create')
+
+
+class UpdateForm(SanicForm):
+    url = StringField('URL', validators=[DataRequired()])
+    submit = SubmitField('Update')
+
+
+@form_blueprint.route('/create', methods=['GET'])
+@login_required
+async def create_link_form(request, user):
+    form = CreateForm(request)
+    content = f"""
+        <div class="container">
+        <form action="" method="POST">
+          <h1 id="form-header">Create a new link</h1>
+          {'<br>'.join(form.csrf_token.errors)}
+          {form.csrf_token}
+          {'<br>'.join(form.endpoint.errors)}
+          <br>
+          <ul>
+          <li>{form.endpoint(size=20, placeholder="Endpoint")}</li>
+          <li>{form.url(size=20, placeholder="URL")}</li>
+          <li>{form.submit}</li>
+          </ul>
+        </form>
+    """
+    base = open('src/templates/base.html', 'r').read()
+    appendix = open('src/templates/forms/create_form.html', 'r').read()
+    return html(base + content + appendix)
+
+
+@form_blueprint.route('/create', methods=['POST'])
+@login_required
+async def create_link_save(request, user):
+    form = CreateForm(request)
+    if not form.validate():
+        return json({'message': 'form invalid'}, status=400)
```

Comment:
D'oh, now, I can see why it was a good idea to have them all in one method... :sweat: Anyway, I don't want you to redo it again... Feel free to discuss my suggestions with me in the future... :pray:

Reply:
It looked good to me, makes errors explicit. 👍